dali/test/tensor_adapter.h

```diff
@@ -0,0 +1,36 @@
+#ifndef DALI_TENSOR_BATCH_ADAPTER_H
+#define DALI_TENSOR_BATCH_ADAPTER_H
+
+#include "dali/pipeline/data/tensor_list.h"
+#include <vector>
+
+namespace dali {
+
+namespace testing {
+
+template<typename UnderlyingDataType>
+class TensorAdapter {
+  static_assert(std::is_fundamental<UnderlyingDataType>::value, "Underlying data type for tensor has to be fundamental");
+public:
+  TensorAdapter(std::vector<UnderlyingDataType> data, std::vector<size_t> shape) noexcept {}
+
+  template<typename Backend, typename T>
+  static dali::TensorList<Backend> ToTensorList(std::vector<TensorAdapter<T>> tensors) noexcept {
```

Comment:
I'd like to ask. Is there a reason for not using const refs when passing containers and other objects around?

Reply:
I did it for clarity, finally I'd change it for `const&`, but here it's just an API PR